site/content/docs/5.0/forms/validation.md

```diff
@@ -354,7 +354,7 @@ If your form layout allows it, you can swap the `.{valid|invalid}-feedback` clas
 
 ## Customizing
 
-Validation states can be customized via Sass with the `$form-validation-states` map. Located in our `_variables.scss` file, this Sass map is looped over to generate the default `valid`/`invalid` validation states. Included is a nested map for customizing each state's color and icon. While no other states are supported by browsers, those using custom styles can easily add more complex form feedback.
+Validation states can be customized via Sass with the `$form-validation-states` map. Located in our `_variables.scss` file, this Sass map is looped over to generate the default `valid`/`invalid` validation states. Included is a nested map for customizing each state's color, icon, tooltip color and focus shadow color. While no other states are supported by browsers, those using custom styles can easily add more complex form feedback.
 
 Please note that we do not recommend customizing these values without also modifying the `form-validation-state` mixin.
```

Comment:
@ffoodd  Wanted to ask about phrase in the docs - `Please note that we do not recommend customizing these values without also modifying the `form-validation-state` mixin.`. Should we keep this? Why need to modify a mixin? Seems should be OK now without modifying. What do you think?

Reply:
Not sure about this either: it's coming from v4 in #27999 (where maps were introduced to handle states).

I think it's still useful since the state key (eg `valid` and `invalid` currently) are used to generate pseudo-classes in `form-validation-state-selector()` mixin.

Any value that's not a form input state pseudo-class would generate invalid and useless selectors.

I guess this is the point of this sentence, but would like @mdo or @MartijnCuppens thought on this.

Reply:
I see, yes, thanks for info, could be reasonable then... But seems there is PR that fixes this possibility to write not correct selector https://github.com/twbs/bootstrap/pull/28447 so that people wanted to add `waiting` https://github.com/twbs/bootstrap/issues/28380 and `warning` / `info` states https://github.com/twbs/bootstrap/pull/26471 

So there is `@if` statement to check if it is `valid` / `invalid` - then use pseudo selector, otherwise - it will add a classname to a selector instead: `is-warning`, `is-waiting`, etc.

Well, we can change this in another PR, the main purpose of this one was to have ability to control hardcoded formulas in SCSS.

Reply:
Indeed, we should try to extend those to check if this sentence is still accurate, bit that may be tackled layer.

Reply:
Yup, let's save this for later. @ydmitry @ffoodd Would either of you want to create an issue for that to track?

Reply:
Added https://github.com/twbs/bootstrap/issues/32082